geo: add pgerror codes for errors.Newf

otan opened the pull request:
Release note (sql change): Added PG error codes to the majority of
spatial related functions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76129)
<!-- Reviewable:end -->

otan committed:
geo: add pgerror codes for errors.Newf

Release note (sql change): Added PG error codes to the majority of
spatial related functions.

rafiss reviewed (approved):
cool stuff!

otan commented:
thxxx

bors r=rafiss
